File: backend/entity/attachment.go
package entity

import (
	"gorm.io/gorm"
)

type Attachment struct {
	gorm.Model
	TargetType string `json:"target_type"` // "thread" | "comment"
	TargetID   uint   `json:"target_id"`
	FileURL    string `json:"file_url"`
	UserID     uint   `json:"user_id"`
	User       *User  `gorm:"foreignKey:UserID" json:"user"`
}

File: frontend/src/interfaces/Attachment.ts
// src/interfaces/Attachment.ts
import type { User } from "./User";
import type { TargetType } from "./Reaction";

export interface Attachment {
  ID: number;
  target_type: TargetType; // "thread" | "comment"
  target_id: number;
  file_url: string;
  user_id: number;
  user?: User;
}

export interface CreateAttachmentRequest {
  target_type: TargetType;
  target_id: number;
  file_url: string;
  user_id: number;
}

export interface UpdateAttachmentRequest {
  ID: number;
  target_type?: TargetType;
  target_id?: number;
  file_url?: string;
}
